### ISSUE TITLE
💅 useHookAtTopLevel is raised incorrectly

### ISSUE THREAD

**Issue by Gojitaa** [biomejs/biome#3365]
### Environment information

```bash
CLI:
  Version:                      1.8.3
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.14.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.7.0"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Linter:
  JavaScript enabled:           true
  JSON enabled:                 true
  CSS enabled:                  false
  Recommended:                  false
  All:                          false
  Enabled rules:
  complexity/noBannedTypes
  suspicious/noCatchAssign
  suspicious/noPrototypeBuiltins
  complexity/useLiteralKeys
  style/useAsConstAssertion
  suspicious/noMisleadingInstantiator
  complexity/noMultipleSpacesInRegularExpressionLiterals
  complexity/noUselessLoneBlockStatements
  complexity/noVoid
  style/noArguments
  suspicious/noDebugger
  suspicious/noUnsafeNegation
  correctness/noConstantCondition
  suspicious/noGlobalAssign
  suspicious/useValidTypeof
  suspicious/noAssignInExpressions
  suspicious/noDuplicateParameters
  correctness/noEmptyPattern
  suspicious/useGetterReturn
  suspicious/noExtraNonNullAssertion
  style/useTemplate
  style/useImportType
  correctness/noUnusedLabels
  complexity/noUselessTernary
  correctness/noUnreachableSuper
  security/noGlobalEval
  correctness/useHookAtTopLevel
  correctness/noUnusedVariables
  correctness/useIsNan
  correctness/noUnsafeOptionalChaining
  correctness/noConstAssign
  correctness/noSwitchDeclarations
  style/useConst
  suspicious/noExplicitAny
  suspicious/noCompareNegZero
  suspicious/noSelfCompare
  complexity/noExcessiveCognitiveComplexity
  complexity/noUselessTypeConstraint
  style/noVar
  suspicious/noAsyncPromiseExecutor
  suspicious/noControlCharactersInRegex
  complexity/noWith
  suspicious/noDoubleEquals
  style/useForOf
  suspicious/noDuplicateClassMembers
  complexity/noExtraBooleanCast
  suspicious/noEmptyInterface
  suspicious/noConfusingLabels
  suspicious/noMisleadingCharacterClass
  correctness/noPrecisionLoss
  suspicious/noRedeclare
  correctness/noSetterReturn
  correctness/noInvalidConstructorSuper
  suspicious/noDuplicateObjectKeys
  suspicious/noUnsafeDeclarationMerging
  complexity/noUselessThisAlias
  correctness/noUnreachable
  suspicious/noFallthroughSwitchClause
  correctness/noInnerDeclarations
  suspicious/noDuplicateCase
  style/noParameterAssign
  style/useShorthandAssign
  correctness/noSelfAssign
  style/noUselessElse
  style/useShorthandFunctionType
  suspicious/noEmptyBlockStatements
  suspicious/noShadowRestrictedNames
  correctness/useValidForDirection
  complexity/noUselessLabel
  complexity/noUselessCatch
  correctness/noInvalidUseBeforeDeclaration
  correctness/noUnsafeFinally
  style/useBlockStatements
  style/useCollapsedElseIf
  complexity/noUselessRename
  correctness/noNonoctalDecimalEscape
  complexity/noUselessConstructor
  style/noRestrictedGlobals
  suspicious/useAwait
  correctness/noUndeclaredVariables
  complexity/noStaticOnlyClass
  style/useSingleVarDeclarator
  style/useWhile
  complexity/useArrowFunction
  style/noCommaOperator
  correctness/useYield
  style/useNumericLiterals
  suspicious/noConsoleLog
  suspicious/noImportAssign
  suspicious/noLabelVar
  style/noNamespace
  correctness/noGlobalObjectCalls
  correctness/noNewSymbol
  correctness/noEmptyCharacterClassInRegex
  suspicious/noClassAssign
  suspicious/noFunctionAssign

Workspace:
  Open Documents:               0
```

### Rule name

useHookAtTopLevel

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=LwAvACAAQQBzAGQALgB0AGUAcwB0AC4AdABzACAAaQBuACAAYQAgAHIAYQBuAGQAbwBtACAAcgBlAGEAYwB0ACAAcAByAG8AagBlAGMAdAAKAAoALwAvACAAMQAuACAAcgBlAHAAcgBvADoAIABuAHAAeAAgAGIAaQBvAG0AZQAgAGMAaABlAGMAawAgAC4ACgAvAC8AIAAgAGUAcgByAG8AcgBzACAAdwBpAHQAaAAgAHUAcwBlAEgAbwBvAGsAQQB0AFQAbwBwAEwAZQB2AGUAbAAgAGEAcwAgAGkAdAAgAHMAZQBlAHMAIABqAGUAcwB0AC4AdQBzAGUARgBhAGsAZQBUAGkAbQBlAHIAcwAgAGEAcwAgAGEAIABoAG8AbwBrAC4ACgAvAC8AIAB0AGgAaQBzACAAbwBuAGUAIABjAGEAbgAnAHQAIABiAGUAIAByAGUAcAByAG8AZAB1AGMAZQBkACAAdwBpAHQAaAAgAGUAcwBsAGkAbgB0ACAACgBqAGUAcwB0AC4AdQBzAGUARgBhAGsAZQBUAGkAbQBlAHIAcwAoACkACgAKAAoALwAvACAAMgAuACAAcgBlAHAAcgBvACwAIABzAG8AbQBlACAAcgBhAG4AZABvAG0AIABmAHUAbgBjAHQAaQBvAG4AIAB0AGgAYQB0ACAAcwB0AGEAcgB0AHMAIAB3AGkAdABoACAAIgB1AHMAZQAiACAAaQBuAHMAaQBkAGUAIABhACAAdABlAHMAdAAgAGYAaQBsAGUAIABmAG8AcgAgAGUAeABhAG0AcABsAGUACgAvAC8AIAB0AGgAaQBzACAAbwBuAGUAIABhAGwAcwBvACAAdABoAHIAbwB3ACAAdQBzAGUASABvAG8AawBBAHQAVABvAHAATABlAHYAZQBsAAoALwAvACAAdABoAGkAcwAgAG8AbgBlACAAYwBhAG4AIABiAGUAIAByAGUAcAByAG8AZAB1AGMAZQBkACAAdwBpAHQAaAAgAGUAcwBsAGkAbgB0ACAAdABvAG8ALgAKAGYAdQBuAGMAdABpAG8AbgAgAHUAcwBlAFMAbwBtAGUAdABoAGkAbgBnACgAKQB7AAoAIAAgAHIAZQB0AHUAcgBuACAAMQAgACsAIAAxAAoAfQAKAAoAdQBzAGUAUwBvAG0AZQB0AGgAaQBuAGcAKAApAA%3D%3D

### Expected result

biome should not throw `useHookAtTopLevel` when using jest.useFakeTimers.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Gojitaa** [comment]
oh I see theres already a bug for this. Closing.
